examples/tutorials/core_api_pytorch_mnist/model_def_checkpoints.py

```diff
@@ -196,7 +213,9 @@ def main(core_context):
     optimizer = optim.Adadelta(model.parameters(), lr=args.lr)
 
     scheduler = StepLR(optimizer, step_size=1, gamma=args.gamma)
-    for epoch in range(1, args.epochs + 1):
+
+    # NEW: Resume training from epochs_completed. This is useful in the case of pausing and resuming an experiment.
+    for epoch in range(epochs_completed, args.epochs + 1):
```

Comment:
two things:

1. you have a bug relative to the old code where the old code started at epoch=1 but this code starts at epoch=0
2. it is way more common to do ranges starting from zero, and you can use the name "epoch_idx" to make it more obvious when reading the code that it's a zero-indexed value rather than a 1-indexed value.  In fact, to make it super easy to read, you can even show the relationship between the `epoch_idx` and the `epochs_completed`:

```python
for epoch_idx in range(epochs_completed, args.epochs):
    train(...)
    epochs_completed = epoch_idx + 1
    steps_completed = epochs_completed * len(train_loader)
    test(...)

    ...
    with core_context.checkpoint.store_path(...):
        with path.joinpath("state").open("w") as f:
            f.write(f"{epochs_completed},{info.trial.trial._id}")
```